webapp/channels/src/utils/notifications.test.ts

```diff
@@ -21,52 +21,67 @@ describe('Notifications.showNotification', () => {
     let store: ReturnType<typeof configureStore>;
 
     beforeEach(() => {
+        // Re-initialize window.Notification so that tests can modify it as needed. By default, everything exists,
+        // we've never requested permissions before, and any request for permissions will be denied.
+        window.Notification = jest.fn(() => ({
+            close: jest.fn(),
+        }));
+        window.Notification.requestPermission = jest.fn(() => Promise.resolve('denied'));
+        window.Notification.permission = 'default';
+
+        // Reset and re-import utils/notifications for every test to reset requestedNotificationPermission
```

Comment:
This comment is just to highlight that this is a great way of testing this. I wasn't aware of any way to reset the value of global state in a file without having to export it

Reply:
Nice! Didn't know this one. It should also work with singletons, right? Like...

```
class Foo {...}

export new Foo();
```

After `resetModules` it should return a new singleton, isn't it?

Reply:
Yeah, it would. I added a `console.log` to the root of `utils/notifications` when I was testing this to confirm that it worked that way, and it ran for every single test